### PR TITLE
Add support for manual indexing

### LINKED ISSUE #176: Allow root file to be excluded and set filename of hierarchy files
I have some questions about this.
For our project this is a breaking change.
We used those files directly in a toctree. Which is not possible anymore due to the `.include` suffix. Is it still possible to do this?

_Originally posted by @IceflowRE in https://github.com/svenevs/exhale/pull/136#issuecomment-1238722511_

Interesting. Until it was brought up I wasn't aware of or intending the original behavior. IIRC it means each directive gets processed twice just one page isn't linked to or something so the default of `.include` is going to be kept.

But it's easy to patch in a config value for `exhale_args` to just allow you to choose the file basename (string) so that you can restore the original behavior. Would that work?

### PR DESCRIPTION
**Pull request by svenevs**
- Allow the root api file generation to be skipped.
- Allow ``.rst.include`` behavior to be modified.
- Release 0.3.6.

Closes #176.  @IceflowRE can you take a look and see if this works for you?  I have not tested this yet.  Outside feedback welcome.

> just an option that wont add the .include suffix

I understand it's a little more inconvenient but see the newly added options in `configs.py`.  Docs on manual indexing in `configs.rst` hopefully gives a little more context, my understanding is you just wanted specific hierarchies but not the unabridged api or orphans (if relevant).

The other thing that came to mind was like `hierarchyFileExtension = ".rst.include"` and then you would only need to set one, but then there might be somebody who only wants say the page or the class hierarchy but not the others (?).

> and do not generate the rootFileName file?

That one should work by setting `rootFileName` to `"EXCLUDE"` now.

-----

Open to suggestions, I'm happy to expose more of the internals for projects to hack around -- exhale is one giant hack.

### CODE REVIEW

**Comment by IceflowRE** [comment]
In my tests the `EXCLUDE` value did not work. 
> sphinx.errors.ConfigError: The given `rootFileName` ("EXCLUDE") did not end with '.rst'; Exhale is reStructuredText only.
The other things works as expected! Thank you.

File: testing/tests/configs.py [hunk 1]
```diff
@@ -392,3 +393,155 @@ def test_orphan_typedef(self):
...
+
+        # Calling tests change the exhale_args and corresponding default parameter,
+        # forcing manual effort deliberately for test correctness.
+        def verify_name(name: str, actual: Path):
+            # These are all in the same directory.
+            self.assertEqual(
+                containment_folder / name,
+                actual,
+                f"Expected {name} and {actual.name} to be the same."
+            )
```
Comment (svenevs) [inline testing/tests/configs.py:452]: `E   AssertionError: WindowsPath('//?/D:/a/exhale/exhale/testing/projects/c[97 chars]ude') != WindowsPath('D:/a/exhale/exhale/testing/projects/cpp_n[93 chars]ude')`: `//?` magic prefix issues on windows.  issues on mac too it seems

**Comment by svenevs** [comment]
> In my tests the EXCLUDE value did not work.

Sorry about that... I knew I needed to add actual tests but hey for an untested stab the implementation was pretty close ^^  That should work as expected.

> The other things works as expected! Thank you.

Thanks for testing, sorry for the delay on finalizing.  The testing framework has ... issues.  This stuff is released on 0.3.6 now.

**Comment by IceflowRE** [comment]
Thanks for the fast response! :D hey for Open Source this is fast!

**Pull request merged**